### PR TITLE
Hide Materialfilter PeriodOnly in ContentNode #4361

### LINKED ISSUE #3862: Hide Materialfilter PeriodOnly in ContentNode
When editing the material for the whole camp, we have a filter on the datatable:

https://github.com/ecamp/ecamp3/assets/7566995/169054d7-21c1-47a4-bb1a-178df1ee1490

This filter can be useful, and we'd like to keep it.

However, when editing a single activity, the filter is also displayed in the material section:

https://github.com/ecamp/ecamp3/assets/7566995/d9e00d9c-0bcd-4891-a157-5b24904c02c3

In this context, displaying period materials does not work and indeed is not a useful interaction. The periodOnly filter should be removed in this view. Note: We still need the delete buttons for all of the material items, but the filter option on this column header should be removed.

### PR DESCRIPTION
**Pull request by MarcoAllenspach**
Hey Manuel

I hope thats not ok....Thank you for your help :)

fix https://github.com/ecamp/ecamp3/issues/3862

### CODE REVIEW

File: frontend/src/components/activity/content/Material.vue [hunk 1]
```diff
@@ -6,6 +6,7 @@
       :layout-mode="layoutMode"
       :material-item-collection="materialItemCollection"
       :disabled="disabled"
+      :disable-period-filter="true"
```
Comment (manuelmeister) [inline frontend/src/components/activity/content/Material.vue]:
```suggestion
      disable-period-filter
```
This is enough, as it coerces to true

File: frontend/src/components/material/MaterialTable.vue [hunk 2]
```diff
@@ -367,10 +376,19 @@ export default {
...
     },
+    // Show filter just if period material is in the list
+    periodFilterEnabled() {
+      return this.materialItemCollection.items.some((item) => item.materialNode === null)
+    },
   },
   mounted() {
     this.clientWidth = this.$el.clientWidth
   },
+  updated() {
```
Comment (manuelmeister) [inline frontend/src/components/material/MaterialTable.vue]: You should probably use the `watch` property and not `updated`.
You could watch the `periodFilterEnabled` and set the periodOnly to false if `periodFilterEnabled` is false.

File: frontend/src/components/material/MaterialTable.vue [hunk 3]
```diff
@@ -157,7 +158,12 @@
               : $tc('components.material.materialTable.reference')
           }}
         </span>
-        <v-icon aria-hidden="true" small :color="periodOnly ? 'primary' : null">
+        <v-icon
+          v-if="periodFilterEnabled"
```
Comment (manuelmeister) [inline frontend/src/components/material/MaterialTable.vue:162]: Why do you remove the icon?

**Review by manuelmeister — CHANGES_REQUESTED** [review]

**Comment by manuelmeister** [comment]
@MarcoAllenspach Thanks for the PR. If you implement the changes, then this PR looks good to me and just requires another review by another eCamp core member.

File: frontend/src/components/material/MaterialTable.vue (hunk 2 above)
Comment (MarcoAllenspach) [reply frontend/src/components/material/MaterialTable.vue]: Done :)

**Review by manuelmeister — APPROVED** [review]
Cool thanks

File: frontend/src/components/material/MaterialTable.vue [hunk 4]
```diff
@@ -252,6 +258,9 @@ export default {
 
     // period Entity for displaying material items within a period (should be null if materialNode is provided)
     period: { type: Object, required: false, default: null },
+
+    // Disable the filter button activity / period
+    disablePeriodFilter: { type: Boolean, required: false, default: false },
```
Comment (pmattmann) [inline frontend/src/components/material/MaterialTable.vue]: This component now has an attribute `disablePeriodFilter` and a computed `periodFilterEnabled`. It took me over 5 minutes to understand why.

Counter suggestion: `hidePeriodFilter` says better what it does.

**Review by pmattmann — CHANGES_REQUESTED** [review]
Otherwise top

File: frontend/src/components/material/MaterialTable.vue (hunk 4 above)
Comment (MarcoAllenspach) [reply frontend/src/components/material/MaterialTable.vue]: You have right the naming is not really happy
I changed it to hidePeriodFilter

**Review by pmattmann — APPROVED** [review]

**Pull request merged**